src/Nix/Eval.hs

```diff
@@ -111,7 +112,9 @@ eval (NSym var) =
     lookupVar var >>= maybe (freeVariable var) (force ?? evaledSym var)
 
 eval (NConstant x)          = evalConstant x
-eval (NStr str)             = evalString str
+eval (NStr str)             = assembleString str >>= \case
+                                Nothing -> evalError @v $ ErrorCall ("failed to evaluate string" :: String)
```

Comment:
The `:: String` qualification isn't needed here, because `String` is the only type that `ErrorCall` accepts.